### PR TITLE
feat(Vibration) - Add Vibration Module

### PR DESCRIPTION
**Pull request by mpodwysocki**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request:

(You can skip this if you're fixing a typo or adding an app to the Showcase.)

Explain the **motivation** for making this change. What existing problem does the pull request solve?

Prefer **small pull requests**. These are much easier to review and more likely to get merged. Make sure the PR does only one thing, otherwise please split it.

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Make sure tests pass on both Travis and Circle CI.

**Code formatting**

Look around. Match the style of the rest of the codebase. See also the simple [style guide](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#style-guide).

### CODE REVIEW

File: Examples/UIExplorer/UIExplorerList.windows.js [hunk 1]
```diff
@@ -49,6 +49,10 @@ var ComponentExamples: Array<UIExplorerExample> = [
 
 const APIExamples = [
   {
+    key: 'VibrationExample',
+    module: require('./VibrationExample'),
+  },
```
Comment (rozele) [inline Examples/UIExplorer/UIExplorerList.windows.js]: Let's keep these in alphabetical order.

File: Libraries/Vibration/VibrationIOS.windows.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+var VibrationIOS = {
+  /**
+   * @deprecated
+   */
+  vibrate: function() {
+    invariant(
+      arguments[0] === undefined,
+      'Vibration patterns not supported.'
```
Comment (rozele) [inline Libraries/Vibration/VibrationIOS.windows.js]: Can you move this back to the ReactWindows/js/Vibration directory?

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using ReactNative.Bridge;
+using Windows.Phone.Devices.Notification;
+
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: Sort (I know, the System namespaces usually come first, but we've done alphabetical everywhere else).

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        internal VibrationModule(bool isMobile)
+        {
+            _isMobile = isMobile;
+        }
+
+        private static bool IsMobile()
+        {
+            var platformFamily = Windows.System.Profile.AnalyticsInfo.VersionInfo.DeviceFamily;
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: Can this be shorted with a using?  At some point we should consider a module / static helper / enum that we can use across all modules and easily swap out the logic for device family detection.

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    /// </summary>
+    public class VibrationModule : NativeModuleBase
+    {
+        private readonly bool _isMobile;
+
+        internal VibrationModule() : this(IsMobile())
+        {
+        }
+
+        internal VibrationModule(bool isMobile)
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: I don't think this constructor is necessary.

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+namespace ReactNative.Modules.Vibration
+{
+    /// <summary>
+    /// Represents the module for vibration.  Note this is only available on Windows Mobile.
+    /// </summary>
+    public class VibrationModule : NativeModuleBase
+    {
+        private readonly bool _isMobile;
+
+        internal VibrationModule() : this(IsMobile())
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: internal APIs still should be documented (and not sure why we need this to be internal).  If there are other modules with internal constructors, we should do a pass to clean that up.

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        public override string Name
+        {
+            get
+            {
+                return "Vibration";
+            }
+        }
+
+        /// <summary>
+        /// Vibrates the device for one second.
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: <paramref name="duration" /> seconds?

**Comment by rozele** [comment]
:+1:

**Comment by rozele** [comment]
Looks good, but please review the comments before merging.

File: Examples/UIExplorer/UIExplorerList.windows.js (hunk 1 above)
Comment (mpodwysocki) [inline Examples/UIExplorer/UIExplorerList.windows.js]: :+1:

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs (hunk 4 above)
Comment (mpodwysocki) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: Fixed as I moved the logic into the `WindowsPlatformHelper` which has a `DeviceFamily` which contains all of the possible devices for UWP.

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs (hunk 5 above)
Comment (mpodwysocki) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: It's there for testability purposes, but if we're touching raw UWP, not much point in that...

File: ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs (hunk 7 above)
Comment (mpodwysocki) [inline ReactWindows/ReactNative/Modules/Vibration/VibrationModule.cs]: Fixed as it should be in milliseconds.

**Pull request merged**